### ISSUE TITLE
[may be helpful] ansible keeps returning errors

### ISSUE THREAD

**Issue by u21h2** [Orange-Cyberdefense/GOAD#79]
## TL;DR
If you also often encounter ansible errors, remember to **check your resource allocation**, which may help.

## nonsense
At the beginning, I set up the lab environment on a 40C Ubuntu machine, and ansible only encountered few errors. I solved it by re-running ansible's docker.
Later, I switched to a Windows machine with i5-12490f (6 cores and 12 threads) to build the lab. The resources I allocated to the virtual machines were 2C4G (still according to the default configuration). At this time, ansible will return errors at almost every step. The step with the most errors is [gathering facts]. It is like all virtual machines are suspended from providing services. I've been struggling with this for days and still can't figure it out.

Until I opened the VirtualBox console, I noticed that the time in each Windows virtual machine was different, with a difference of up to two minutes. The strange thing is that they start at the same time, and some machines will slow down for a few minutes after a period of time. This inspires me: Is there a situation where resources are preempted from each other?
I adjusted the configuration of the virtual machine to 1C via Vagrantfile. At this time, it is observed that the time of each virtual machine is consistent, and the execution of ansible is also very smooth.
It was my poor OS knowledge that caused me to waste a lot of time on this. :D

**Comment by tyfhxn** [comment]
it's good

**Comment by pr0b3r7** [comment]
I added 4C6G to each box and that greatly reduced the number of errors in ansible on provisioning time
